src/components/colorscale/calc.js

```diff
@@ -84,9 +86,9 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
     doUpdate(autoAttr, (auto !== false || (min === undefined && max === undefined)));
 
     if(container.autocolorscale) {
-        if(min * max < 0) scl = scales.RdBu;
-        else if(min >= 0) scl = scales.Reds;
-        else scl = scales.Blues;
+        if(min * max < 0) scl = gd._fullLayout.colorscale.diverging;
```

Comment:
minor :cow2: , usually when a method uses `fullLayout` from `gd` we put a

```js
var fullLayout = gd._fullLayout;
```

just below the `function (`